### PR TITLE
"bad-request-handled-validation"

### PR DESCRIPTION
**Pull request by MallanagoudaB**

### CODE REVIEW

File: src/module/userRoleExtension/validator/v1.js [hunk 1]
```diff
@@ -27,12 +27,14 @@ module.exports = (req) => {
 		update: function () {
 			req.checkBody('title').exists().withMessage('required title')
 			req.checkParams('_id').exists().withMessage('required _id')
+			req.checkParams('_id').exists().isMongoId().withMessage('Invalid MongoDB ID')
```
Comment (VISHNUDAS-tunerlabs) [inline src/module/userRoleExtension/validator/v1.js:30]: @MallanagoudaB message can be invalid userRoleExtension id

File: src/module/entities/validator/v1.js [hunk 2]
```diff
@@ -44,19 +58,23 @@ module.exports = (req) => {
...
 			req.checkParams('_id').exists().withMessage('required Entity type')
+			req.checkParams('_id').exists().isMongoId().withMessage('Invalid MongoDB ID')
 		},
 		subEntityListBasedOnRoleAndLocation: function () {
 			req.checkParams('_id').exists().withMessage('required state location id')
 		},
 		details: function () {
-			req.checkParams('_id').exists().withMessage('required state location id')
+			req.checkParams('_id').exists().withMessage('required id')
+			req.checkParams('_id').exists().isMongoId().withMessage('Invalid MongoDB ID')
```
Comment (VISHNUDAS-tunerlabs) [inline src/module/entities/validator/v1.js:68]: message can be invalid entity id @MallanagoudaB

File: src/module/entities/validator/v1.js [hunk 3]
```diff
@@ -44,19 +58,23 @@ module.exports = (req) => {
...
 		},
 		list: function () {
 			req.checkQuery('type').exists().withMessage('required type')
 			req.checkParams('_id').exists().withMessage('required entity id')
+			req.checkParams('_id').exists().isMongoId().withMessage('Invalid MongoDB ID')
```
Comment (VISHNUDAS-tunerlabs) [inline src/module/entities/validator/v1.js:73]: @MallanagoudaB message can be invalid entity id

File: src/module/entities/validator/v1.js [hunk 4]
```diff
@@ -44,19 +58,23 @@ module.exports = (req) => {
...
 		},
 		relatedEntities: function () {
 			req.checkParams('_id').exists().withMessage('required Entity id')
+			req.checkParams('_id').exists().isMongoId().withMessage('Invalid MongoDB ID')
```
Comment (VISHNUDAS-tunerlabs) [inline src/module/entities/validator/v1.js:77]: @MallanagoudaB message can be invalid entity id

File: src/module/entityTypes/validator/v1.js [hunk 5]
```diff
@@ -20,6 +20,7 @@ module.exports = (req, res) => {
 		},
 		update: function () {
 			req.checkParams('_id').exists().withMessage('required _id')
+			req.checkParams('_id').exists().isMongoId().withMessage('Invalid MongoDB ID')
```
Comment (VISHNUDAS-tunerlabs) [inline src/module/entityTypes/validator/v1.js:23]: message can be invalid entity type id

File: src/controllers/v1/entityTypes.js [hunk 6]
```diff
@@ -263,7 +263,9 @@ module.exports = class EntityTypes extends Abstract {
 
 					input.push(null)
 				} else {
-					throw CONSTANTS.apiResponses.PROJECT_FAILED
+					const error = new Error(CONSTANTS.apiResponses.PROJECT_FAILED)
```
Comment (VISHNUDAS-tunerlabs) [inline src/controllers/v1/entityTypes.js:266]: @MallanagoudaB what is PROJECT_FAILED response?

File: src/controllers/v1/entityTypes.js [hunk 7]
```diff
@@ -241,7 +241,7 @@ module.exports = class EntityTypes extends Abstract {
 				const newEntityTypeData = await entityTypesHelper.bulkCreate(entityTypesCSVData, req.userDetails)
 
 				// Check if entity types were created successfully
-				if (newEntityTypeData.length > 0) {
+				if (newEntityTypeData.length > 0 && newEntityTypeData[0].status === CONSTANTS.apiResponses.SUCCESS) {
```
Comment (VISHNUDAS-tunerlabs) [inline src/controllers/v1/entityTypes.js:244]: @MallanagoudaB how are we notifying user if some entity creation failed ?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
reiewed on Dec 11

File: src/controllers/v1/entityTypes.js (hunk 7 above)
Comment (MallanagoudaB) [reply src/controllers/v1/entityTypes.js:244]: This not what we discussed @VISHNUDAS-tunerlabs 
Its another ticket

File: src/generics/constants/api-responses.js [hunk 8]
```diff
@@ -56,4 +55,5 @@ module.exports = {
 	USER_ROLE_UPDATED: 'USER_ROLE_UPDATED',
 	USER_ROLE_DELETED: 'USER_ROLE_DELETED',
 	ENTITY_UPDATED: 'ENTITY_UPDATED',
+	ENTITYTYPE_CREATION_FAILED: 'ENTITYTYPE CREATION FAILED',
```
Comment (VISHNUDAS-tunerlabs) [inline src/generics/constants/api-responses.js:58]: @MallanagoudaB ENTITY_TYPE_CREATION_FAILED change the constant like this

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

File: src/generics/constants/api-responses.js [hunk 9]
```diff
@@ -56,4 +55,5 @@ module.exports = {
 	USER_ROLE_UPDATED: 'USER_ROLE_UPDATED',
 	USER_ROLE_DELETED: 'USER_ROLE_DELETED',
 	ENTITY_UPDATED: 'ENTITY_UPDATED',
+	ENTITY_TYPE_CREATION_FAILED: 'ENTITY TYPE CREATION FAILED',
```
Comment (VISHNUDAS-tunerlabs) [inline src/generics/constants/api-responses.js:58]: @MallanagoudaB Where are you using  this? have you made changes all the places we use this as well?

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

**Pull request merged**